lib/rust/shapely/macros/src/tagged_enum.rs

```diff
@@ -0,0 +1,298 @@
+use crate::prelude::*;
```

Comment:
I haven't looked into implementation but the docs suggest that this file largely duplicates functionality of `to_variant_types` from the previous AST implementation.

Reply:
It is similar, yes. However both implementations define things optimized for the given AST implementation.